src/container/streamAnalyticsManager.ts

```diff
@@ -136,32 +138,42 @@ export class StreamAnalyticsManager {
 
             let retryTimes = 0;
             while (true) {
-                await this.sleep(1000);
+                await this.sleep(2000);
 
                 const jobInfoResult = await request.get(operationResultUrl, {
                     auth: {
                         bearer: aadAccessToken,
                     },
+                    resolveWithFullResponse: true,
                 });
 
-                if (!jobInfoResult) {
+                if (token.isCancellationRequested) {
+                    throw new UserCancelledError();
+                }
+
+                if (jobInfoResult.statusCode === 202) {
                     if (retryTimes < this.MaximumRetryCount) {
                         retryTimes++;
                         continue;
                     } else {
                         throw new Error(Constants.queryASAJobInfoFailedMsg);
                     }
-                }
-
-                const result = JSON.parse(jobInfoResult);
-                if (result.status === "Succeeded") {
-                    const info = JSON.parse(result.manifest);
-                    return info;
+                } else if (jobInfoResult.statusCode === 200) {
+                    const result = JSON.parse(jobInfoResult.body);
+                    if (result.status === "Succeeded") {
+                        const info = JSON.parse(result.manifest);
+                        return info;
+                    } else {
+                        throw new Error(result.error.message);
+                    }
                 } else {
-                    throw new Error(result.error.message);
+                    throw new Error("http status code: " + jobInfoResult.statusCode);
                 }
             }
         } catch (error) {
+            if (error instanceof UserCancelledError) {
```

Comment:
How about change to if not instance of UserCancelledError, do formatting, So "throw error" common logic could be shared

Reply:
Updated